### DISCUSSION TITLE
beep when pressing a button - image keyboard response plugin

### DISCUSSION THREAD

**Discussion by AnneVogt** [jspsych/jsPsych#851] [Q&A]
Dear all,
I am using the image keyboard response plugin for some of my trials with these parameters:

```js
var buttonpress_trial = {
        type: "image-keyboard-response",
        stimulus: jsPsych.timelineVariable("stimulus"),
        choices: ["q", "p"],
        trial_duration: 2000,
        response_ends_trial: false,
        data: jsPsych.timelineVariable("data"),
        on_finish: function(data){
            if(data.key_press == jsPsych.pluginAPI.convertKeyCharacterToKeyCode(data.target_response)){
            data.correct = true;
            } else {
            data.correct = false;
            }
        }       
};
```

During piloting we noticed that on one Mac with Safari a beep was played everytime the participant pressed the button although this was not intended. We even tell participants in the instruction that the picture will remain on screen for two seconds no matter when they press the button. The logfile looked ok, it was correctly recorded whether the participants pressed the correct button or not. With a Windows PC using either Chrome or Firefox this did not happen. I made no changes to the plugin and therefore, I am at loss how to trace the problem.
Hope somebody has an idea.
Best,
Anne

**Comment by jodeleeuw** [comment]
I don't have a mac available right now to test this out. @becky-gilbert or @vijaymarupudi, do you have a mac?

It might be related to [this](https://stackoverflow.com/questions/57726300/safari-error-sound-when-pressing-arrow-keys-to-navigate-gallery), but we do use `preventDefault()` in the keyboard handler:

https://github.com/jspsych/jsPsych/blob/83980085ef604c815f0d97ab55c816219e969b84/jspsych.js#L2062

**Reply by vijaymarupudi** [reply]
I'm on Linux :(

**Reply by becky-gilbert** [reply]
I have access to a Mac and will test this soon!
